conda/exceptions.py

```diff
@@ -531,30 +575,60 @@ class CondaHTTPError(CondaError):
     def __init__(self, message, url, status_code, reason, elapsed_time, response=None,
                  caused_by=None):
         from .common.url import maybe_unquote
-        _message = dals("""
-        HTTP %(status_code)s %(reason)s for url <%(url)s>
-        Elapsed: %(elapsed_time)s
-        """)
-        cf_ray = getattr(response, 'headers', {}).get('CF-RAY')
-        _message += "CF-RAY: %s\n\n" % cf_ray if cf_ray else "\n"
-        message = _message + message
 
+        # if response includes a valid json body we prefer the reason/message defined there
+        try:
+            body = response.json()
+        except (AttributeError, JSONDecodeError):
+            body = {}
+        else:
+            reason = body.get("reason", None) or reason
+            message = body.get("message", None) or message
+
+        # standardize arguments
+        url = maybe_unquote(url)
         status_code = status_code or '000'
         reason = reason or 'CONNECTION FAILED'
+        if isinstance(reason, str):
+            reason = reason.upper()
         elapsed_time = elapsed_time or '-'
+        if isinstance(elapsed_time, timedelta):
+            elapsed_time = str(elapsed_time).split(":", 1)[-1]
+
+        # extract CF-RAY
+        try:
+            cf_ray = response.headers["CF-RAY"]
+        except (AttributeError, KeyError):
+            cf_ray = ""
+        else:
+            cf_ray = f"CF-RAY: {cf_ray}\n"
+
+        # compose message
+        message = (
+            dals(
+                f"""
+                HTTP {status_code} {reason} for url <{url}>
+                Elapsed: {elapsed_time}
+                {cf_ray}
+                """
+            )
+            # since message may include newlines don't include in f-string/dals above
+            + message
+        )
 
         from .auxlib.logz import stringify
         response_details = (stringify(response, content_max_len=1024) or '') if response else ''
```

Comment:
We should also fix the bug I discovered above here and get that import statement out of here if possible.